automatilib/cola_auth/i_dot_ai_cola_views.py

```diff
@@ -0,0 +1,152 @@
+import logging
+import re
+from abc import abstractmethod
+from urllib.parse import unquote
+
+import requests
+from django.conf import settings
+from django.contrib.auth import authenticate, get_user_model, login, logout
+from django.http import (
+    HttpRequest,
+    HttpResponse,
+    HttpResponseBadRequest,
+    HttpResponseServerError,
+)
+from django.shortcuts import redirect
+from django.urls import reverse
+from jose import jwt
+from jose.exceptions import ExpiredSignatureError, JWTClaimsError, JWTError
+
+from automatilib.core.i_dot_ai_utils import MethodDispatcher
+
+LOGGER = logging.getLogger(__name__)
+
+User = get_user_model()
+
+
+def get_cola_cognito_user_pool_jwk() -> tuple[dict, str]:
+    cola_issuer = f"https://cognito-idp.{settings.AWS_REGION_NAME}.amazonaws.com/{settings.COLA_COGNITO_USER_POOL_ID}"
+    response = requests.get(f"{cola_issuer}/.well-known/jwks.json", timeout=5)
+    assert response.status_code == 200
```

Comment:
What happens if it is not 200? What logging are we going to need for our records and what message is passed on to the user?

Reply:
I've added a clause to return None in this instance so the view can handle it. We can hopefully handle all errors in the view. Not sure how we want to display these to the user though as we don't have any templates in this project (yet?)